File: src/#513/solution.go
package solution

type TreeNode struct {
	Val   int
	Left  *TreeNode
	Right *TreeNode
}

func findBottomLeftValue(root *TreeNode) int {
	maxLevel := 0
	res := 0
	helper(root, 0, &maxLevel, &res)
	return res
}

func helper(root *TreeNode, level int, maxLevel, res *int) {
	if root == nil {
		return
	}
	if level >= *maxLevel {
		*res = root.Val
		*maxLevel++
	}
	helper(root.Left, level+1, maxLevel, res)
	helper(root.Right, level+1, maxLevel, res)
}

File: src/#513/solution.js
function TreeNode(val, left, right) {
  this.val = val === undefined ? 0 : val;
  this.left = left === undefined ? null : left;
  this.right = right === undefined ? null : right;
}

var findBottomLeftValue = function (root) {
  let maxLevel = 0;
  let res = 0;

  helper(root, 0);

  return res;

  function helper(root, level) {
    if (root === null) {
      return;
    }

    if (level >= maxLevel) {
      res = root.val;
      maxLevel++;
    }

    helper(root.left, level + 1);
    helper(root.right, level + 1);
  }
};
